### PR TITLE
fix wrong execution job on DELETE API - for master

### PR DESCRIPTION
**Pull request by clemensutschig**
closes jenkins shared lib #790 (https://github.com/opendevstack/ods-jenkins-shared-library/issues/790) for master

### CODE REVIEW

File: src/main/java/org/opendevstack/provision/services/JenkinsPipelineAdapter.java [hunk 1]
```diff
@@ -592,8 +595,18 @@ private static void validateComponentName(
       logger.debug("Calling job {} for project {}", job.getId(), projectKey);
       ExecutionsData data = prepareAndExecuteJob(job, options, webhookProxySecret);
       logger.info("Result of cleanup: {}", data.toString());
+      ExecutionJob result = new ExecutionJob();
+      result.setName(data.getJobName());
+      result.setUrl(data.getPermalink());
+      List<ExecutionJob> currentJobs = project.getLastExecutionJobs();
+      if (currentJobs == null) {
+        currentJobs = new ArrayList<ExecutionJob>();
+        project.setLastExecutionJobs(currentJobs);
```
Comment (stitakis) [inline src/main/java/org/opendevstack/provision/services/JenkinsPipelineAdapter.java:604]: This line changes the object `project`. It is not clear from the caller method point of view the effect on the object roject. Therefore it is a side effect which generally should be avoid.

I'd tend to put the logic that modify object `project` in the `ProjectAPIController`. To do this, some few more changes would need to be done, like modifying the interface method  `IServiceAdapter.cleanUp(...)` to give back the execution job details.
 
If this change is urgent then I'd suggest at least to rename the method to express the side-effect in the object `project`. The name would be then e.g. : `runDeleteAdminJobAndSetAsLastExecutionJobToProject`

Comment (stitakis) [reply src/main/java/org/opendevstack/provision/services/JenkinsPipelineAdapter.java:604]: Let us reduce the number of code lines to try  to make it easier to read, e.g. to:

```
      ExecutionJob result = new ExecutionJob();
      result.setName(data.getJobName());
      result.setUrl(data.getPermalink());

      if (project.getLastExecutionJobs() == null) {
        project.setLastExecutionJobs(new ArrayList<ExecutionJob>());
      }
      project.getLastExecutionJobs().add(result);
```
Looks better?

File: src/main/java/org/opendevstack/provision/services/JenkinsPipelineAdapter.java [hunk 2]
```diff
@@ -592,8 +595,18 @@ private static void validateComponentName(
...
+      }
+      currentJobs.add(result);
       return Collections.emptyMap();
     } catch (RuntimeException | IOException e) {
+      e.printStackTrace();
```
[the remark this answers is not in the document]
Comment (stitakis) [reply src/main/java/org/opendevstack/provision/services/JenkinsPipelineAdapter.java]: done

File: src/main/java/org/opendevstack/provision/services/JenkinsPipelineAdapter.java (hunk 1 above)
Comment (stitakis) [reply src/main/java/org/opendevstack/provision/services/JenkinsPipelineAdapter.java]: I've renamed the method to the suggested name `runDeleteAdminJobAndSetAsLastExecutionJobToProject` and also reorder the code to make it more readable

**Review by stitakis — APPROVED** [review]

**Comment by clemensutschig** [comment]
4.x yes - 3.x no - imho

**Comment by stitakis** [comment]
ok, I'll backport this tomorrow to `4.x`

**Pull request merged**